File: algojava/L0003MaxProfit.java
public class L0003MaxProfit{
    public static void main(String [] args){
       int[] prices = {7,1,5,4,6,3};
       System.out.println(maxProfit(prices));
    }
    public static int maxProfit(int[] prices) {
        int lowPrice = prices[0];
        int prevDiff = 0;
        int currDiff = 0;
        for (int price : prices) {
            if (lowPrice > price) {
                lowPrice = price;
            }
        }
        for (int i = 0; i < prices.length; i++) {
            for (int j = i+1; j < prices.length; j++){
                if(prices[i] < prices[j]){
                    currDiff = prices[j] - prices[i];
                }
                if(currDiff > prevDiff){
                    prevDiff = currDiff;
                }
            }
            if(prices[i] == lowPrice){
                return prevDiff;
            }
        }

    return prevDiff;
    }
}

File: typeScript/L0003MaxProfit.ts
let prices:number[]= [7,1,5,3,6,4];
console.log("maxProfit", maxProfit(prices));
function maxProfit(prices: number[]): number {
    let lowPrice = prices[0];
    for(let i = 1 ; i < prices.length; i ++) {
        if(lowPrice > prices[i]){
            lowPrice = prices[i];
        }
    }
    let prevDiff = 0;
    let currDif = 0;
    for (let i = 0; i < prices.length; i++) { //0=7,1=1
        for (let j = i+1; j < prices.length; j++){//2=5,3=3,4=6,5=4
            if(prices[i] < prices[j]){//7<1,5,3,6,4 false 1<5 true
                currDif = prices[j] - prices[i]; //4,2,5,3
            }
            if(currDif > prevDiff){ //4 > 0,2 > 4 false,5 > 4,3 > 5 false
                prevDiff = currDif;//prevDiff= 4,5
            }
        }
        if(prices[i] === lowPrice){
            return prevDiff;
        }
    }
    return prevDiff;
}